### PR TITLE
=per #18228 Convert Persistence schema evolution guide to Java

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #18228

File: akka-docs/rst/java/persistence-schema-evolution.rst [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+a complete guide, so feel free to adapt these techniques depending on your serializer's capabilities
+and/or other domain specific limitations.
+
+.. _add-field-java:
+
+Add fields
+----------
+
+**Situation:**
+You need to add a field to an existing message type. For example, a ``SeatReservation(letter:String, row:Int)`` now
```
Comment (ktoso) [inline akka-docs/rst/java/persistence-schema-evolution.rst]: Maybe more Java like like so:

```
SeatReservation(String lette, Integer row)
```

File: akka-docs/rst/java/persistence-schema-evolution.rst [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+**Solution:**
+Adding fields is the most common change you'll need to apply to your messages so make sure the serialization format
+you picked for your payloads can handle it apropriately, i.e. such changes should be *binary compatible*.
+This is easily achieved using the right serializer toolkit – we recommend something like `Google Protocol Buffers`_ or
+`Apache Thrift`_ however other tools may fit your needs just as well – picking a serializer backend is something
+you should research before picking one to run with. In the following examples we will be using protobuf, mostly because
+we are familiar with it, it does its job well and Akka is using it internally as well.
+
+While being able to read messages with missing fields is half of the solution, you also need to deal with the missing
+values somehow. This is usually modeled as some kind of default value, or by representing the field as an ``Option[T]``
```
Comment (ktoso) [inline akka-docs/rst/java/persistence-schema-evolution.rst]: `Optional<T>`

File: akka-docs/src/main/protobuf/FlightAppModels.proto [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+// FlightAppModels.proto
+option java_package = "docs.persistence.proto";
+option optimize_for = SPEED;
+
+message SeatReserved {
+  required string letter   = 1;
+  required uint32 row      = 2;
+  optional string seatType = 3; // the new field
+}
+//#protobuf-read-optional-proto
```
Comment (ktoso) [inline akka-docs/src/main/protobuf/FlightAppModels.proto]: good idea to separate it out

**Comment by ktoso** [comment]
LGTM just 2 minor fixups to remove scala-ish snippets from the rst, otherwise thanks!

Heroic Effort!

**Comment by patriknw** [comment]
@ktoso thanks, I fixed the two minor

**Pull request merged**